src/guacd/connection.c

```diff
@@ -278,10 +278,18 @@ static int guacd_route_connection(guacd_proc_map* map, guac_socket* socket) {
         proc = guacd_proc_map_retrieve(map, identifier);
         new_process = 0;
 
-        /* Warn if requested connection does not exist */
-        if (proc == NULL)
-            guacd_log(GUAC_LOG_INFO, "Connection \"%s\" does not exist.",
-                    identifier);
+        /* Warn and ward off client if requested connection does not exist */
+        if (proc == NULL) {
+            char message[2048];
+
+            snprintf(message, sizeof(message),
+                    "Connection \"%s\" does not exist", identifier);
+
+            guacd_log(GUAC_LOG_INFO, message);
```

Comment:
`guacd_log()` will interpret `message` here as a printf-style format string:

https://github.com/apache/guacamole-server/blob/ca1fbd5e980392dc91b52ddff473698035d8f357/src/guacd/log.h#L45-L49

To do this safely, the call would need to be:

```c
guacd_log(GUAC_LOG_INFO, "%s", message);
```

However, given that the connection is failing, I think `GUAC_LOG_ERROR` would be more appropriate.

Reply:
Oh, I missed that -- so I can basically safely log a format string and skip constructing the message entirely? i.e.
```guacd_log(GUAC_LOG_INFO, "Connection \"%s\" does not exist", identifier);```

As for the level, this isn't a runtime error so much as a client error though, akin to HTTP 400 "bad request" or 404 "not found", which to me sounds more like an informative message if any; If you still think it's appropriate to log this an error, however, please ping me and I'll adjust the PR.

Reply:
> Oh, I missed that -- so I can basically safely log a format string and skip constructing the message entirely? i.e. `guacd_log(GUAC_LOG_INFO, "Connection \"%s\" does not exist", identifier);`

Yep!

> As for the level, this isn't a runtime error so much as a client error though, akin to HTTP 400 "bad request" or 404 "not found", which to me sounds more like an informative message if any; ...

Good point. I agree it makes sense for this to remain at the info level, with any further handling of the client error naturally up to the client.

Reply:
Woah, it's been a while. I am no longer with my previous employer and have no real-world way of testing this (nor even a toolchain to build it with...). Hopefully it's still somewhat useful as-is.

Reply:
Ah - right. I now remember something about that from your other PR...

I'll build off what you've done here and get this closed out.